### DISCUSSION TITLE
Add "Indexed" toggle to field setup

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#5503] [Feature Requests]
Gives app/api control over which fields are indexed in the database, for improved performance.

**Comment by sxsws** [comment]
db systems differ in how they handle indexes. Generally, 
- indexes need to be defined on the collection, not the field because indexes can include more than one field, e.g. searching for last name, first name can be grouped into one index. Column priority is important
- the ability to give the index a name is important in a database-first implementation
- partial indexes allow specifying a where clause for the index

and the kickers:
- some systems allow functional indexes, whilst others do not. e.g. `CREATE INDEX idx_foo on tblname ((upper(colName)))` is valid on postgres and mysql but not mariadb (bad example as mysql/mariadb is case insensitive by default!).
- some systems allow indexes on jsonb field data (as opposed to the whole field e.g. you could build an index for something like `.property1.subprop` that would index searches for just the `property.subprop` values in the jsonb field). This feature is fairly new to mariadb compared to mysql.

So, when creating an index, it'll need the properties: 
- a list of `columns` that are sortable to specify the order of them in the index,
- optional `name` which is auto defined if null,
- optional `where clause` for partial indexes, 
- optional function definition, 
- and in the case of jsonb fields, a definition of the index

So, I'm thinking perhaps a separate section on the collections page for indexes, rather than a toggle on the field? I understand a toggle per field may be easier to implement but it limits the power of indexes to really basic things.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
